classes/class.ilAdobeConnectDfnXMLAPI.php

```diff
@@ -79,33 +78,35 @@ public function addUser($login, $email, $pass, $first_name, $last_name, $session
 	 * @param string $session
 	 * @return bool|string
 	 */
-	public function searchUser($login, $session)
+	public function searchUser($login, $session = NULL)
 	{
 		global  $DIC;
 		$ilLog = $DIC->logger()->root();
 
 		$url = $this->getApiUrl(array(
 			'login'     => $login,
 			'action' 	=> 'lms-user-exists',
-			'session' 	=> $session
+			'session' => $session
 		));
-		$xml = simplexml_load_file($url);
-
-		if($xml->status['code'] == 'ok')
+		
+		$xml = $this->sendRequest($url);
+		if($xml instanceof \SimpleXMLElement)
 		{
-			$list = $xml->{'principal-list'};
-			
-			$id = (string)$list->principal['principal-id'];
+			if($xml->status['code'] == 'ok')
+			{
+				$list = $xml->{'principal-list'};
+				$id = (string)$list->principal['principal-id'];
 
-			return $id;
+				return $id;
+			}
```

Comment:
We have no logging anymore if `($xml->status['code'] == 'ok')` is `false`.

Reply:
done